src/sas/qtgui/Plotting/PlotterBase.py

```diff
@@ -17,6 +17,49 @@
 
 DEFAULT_CMAP = mpl.cm.jet
 
+class CustomToolbar(NavigationToolbar):
+    def __init__(self, canvas, parent=None):
+        super().__init__(canvas, parent)
+        self.parent = parent
+        self.addSeparator()
+        self.add_custom_button()
+
+
+
+    def add_custom_button(self):
+        # I have been told that a Button is better
+        # But all NavitaionToolbar are interactions are Actions
+        # This way all can be called with:
+        #   self._actions['xxx']
+        custom_icon = QtGui.QIcon()  # You can load an icon here if you want e.g., QtGui.QIcon("path/to/icon.png")
+        custom_action = QtGui.QAction(custom_icon, "Fitting", self)
+        custom_action.setToolTip("Click to send data to Fitting")
+        custom_action.triggered.connect(self.sendToFitting)
+        self.addAction(custom_action)
+        self._actions['fitting'] = custom_action
+        self._actions['fitting'].setVisible(False)
+
+    def sendToFitting(self):
+        current_file_name: str = self.parent.lable_name
+        search_name = current_file_name
+
+        def find_name(model, target_name: str, column: int=0) -> int:
+            for row in range(model.rowCount()):
+                for row2 in range(model.item(row, column).rowCount()):
+                    tmp = model.item(row, column).child(row2, column)
+                    if tmp.text() == target_name:
+                            return row, row2
+
+        model = self.parent.parent.model
+        row_index_parent, row_index_child = find_name(model, search_name)
+        data_dir = self.parent.parent.model.item(row_index_parent)
+        data = data_dir.child(row_index_child)
+        new_item = self.parent.parent.parent.filesWidget.cloneTheory(data)
+        model.beginResetModel()
+        model.appendRow(new_item)
+        model.endResetModel()
+        self.parent.parent.parent.filesWidget.sendData(None, [new_item])
```

Comment:
Doing all these operations in PlotterBase, by referencing `parent.parent.parent....` is smelly.
The best idea is to implement something like `freezeDataWithName` in DataExplorer, so no such object path traverse is taking place.
Here, you would just find out the name (line 43) and pass it to a signal, which DataExplorer would act on and do all these operations locally.